README.md

```diff
@@ -1,18 +1,128 @@
+  * `claimable` - tracks users' claimable rewards and current reward index for each `ETHStaticAToken`
+  * `token_bridge` - is responsible for:
+    * bridging the staticATokens to and from L2. Minting and burning ETHStaticATokens on message from L1. 
+    * bridging rewAAVE token back to L1
+    * updating `rewards_index` for each ETHStaticAToken on message from L1 
+  * `rewAAVE` - a very basic ERC20 to represent the rewards on L2
+  *  `proxy` - a generic implementation of a proxy in starknet
+
+## ETHStaticATokens on L2
+
+Natively, on Aave aTokens grow in balance, not in value. To be able to create this kind of model, it is important to wrap them before bridging, converting them in a token that grows in value, not in balance.
+
+ETHStaticATokens are an implementation of the wrapped aTokens that will continuously increase in value on Starknet because they are backed by the increasing staticATokens amounts locked in the bridge contract on Ethereum. The ETHStaticATokens can then be bridged back to staticATokens.
+
+## Bridging staticATokens from L1<>L2
+
+### Transfer L1->L2: 
+
+Users can either bridge their staticAToken to L2 by calling `deposit()` on `TokenBridge`, or deposit the underlying asset of the staticAToken directly by calling `depositUnderlying()`.
+
+Calling `deposit` will result in the following:
+
+- staticATokens will be transfered from the user account to the L1 bridge
+- A `deposit` event will be emitted with the L1 token address, the recipient address on L2, and the amount
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+Calling `depositUnderlying` will result in the following:
+
+Users will have to approve the bridge to spend the underlying `asset` tokens or `aTokens`, depending on the provided value for `fromAsset` argument when depositing.
+
+- the underlying asset of the staticAToken or the aToken (depending on the user's input )  will be transfered from the user account to the L1 bridge
+- the bridge will then deposit the aTokens/asset on the staticAToken
+- new staticATokens will be minted to the bridge
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+
+### Transfer L2->L1:
+
+To bridge their staticATokens back to L1, users need to call `initiate_withdraw()` on Starknet `token_bridge`. 
+
+Calling `initiate_withdraw` will result in the following:
+
+- The amount withdraw of ETHStaticAToken will be burned by the bridge
+- A message will be sent to L1 with the L1 token address, the l1 recipient, and the amount
+
+(@TODO: add more info on how we bridge the current rewards index when withdrawing..)
```

Comment:
Just needs to read:

We track the reward index on departure of the static token and send the rewards accrued during the bridging process to the recipients address.